go/api/scripting_and_function_commands_test.go

```diff
@@ -746,3 +746,136 @@ func ExampleGlideClusterClient_FunctionKillWithRoute() {
 	// Output:
 	// Expected error: An error was signalled by the server: - NotBusy: No scripts in execution right now.
 }
+
+func ExampleGlideClient_InvokeScript() {
+	client := getExampleGlideClient()
+
+	// Create a simple Lua script that returns a string
+	script := options.NewScript("return 'Hello from Lua'")
+
+	// Execute the script
+	result, err := client.InvokeScript(*script)
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+		return
+	}
+
+	fmt.Println(result)
+
+	// Output:
+	// Hello from Lua
+}
+
+func ExampleGlideClusterClient_InvokeScript() {
+	client := getExampleGlideClusterClient()
+
+	// Create a simple Lua script that returns a number
+	script := options.NewScript("return 123")
+
+	// Execute the script
+	result, err := client.InvokeScript(*script)
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+		return
+	}
+
+	fmt.Println(result)
+
+	// Output:
+	// 123
+}
+
+func ExampleGlideClient_InvokeScriptWithOptions() {
+	client := getExampleGlideClient()
+
+	// Create a Lua script that uses keys and arguments
+	scriptText := `
+		local key = KEYS[1]
+		local value = ARGV[1]
+		redis.call('SET', key, value)
+		return redis.call('GET', key)
+	`
+	script := options.NewScript(scriptText)
+
+	// Create a unique key for testing
+	testKey := "test-key-" + uuid.New().String()
+
+	// Set up script options with keys and arguments
+	scriptOptions := options.NewScriptOptions().
+		WithKeys([]string{testKey}).
+		WithArgs([]string{"Hello World"})
+
+	// Execute the script with options
+	result, err := client.InvokeScriptWithOptions(*script, *scriptOptions)
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+		return
+	}
+
+	fmt.Println(result)
+
+	// Output:
+	// Hello World
+}
+
+func ExampleGlideClusterClient_InvokeScriptWithOptions() {
+	client := getExampleGlideClusterClient()
+
+	// Create a Lua script that performs calculations with arguments
+	scriptText := `
+		local a = tonumber(ARGV[1])
+		local b = tonumber(ARGV[2])
+		return a + b
+	`
+	script := options.NewScript(scriptText)
+
+	// Set up script options with arguments
+	scriptOptions := options.NewScriptOptions().
+		WithArgs([]string{"10", "20"})
+
+	// Execute the script with options
+	result, err := client.InvokeScriptWithOptions(*script, *scriptOptions)
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+		return
+	}
+
+	fmt.Println(result)
+
+	// Output:
+	// 30
+}
+
+func ExampleGlideClusterClient_InvokeScriptWithClusterOptions() {
+	client := getExampleGlideClusterClient()
+
+	// Create a Lua script.
+	scriptText := "return 'Hello'"
+
+	script := options.NewScript(scriptText)
+
+	// Set up cluster script options
+	clusterScriptOptions := options.NewClusterScriptOptions()
+
+	// Set the route
+	route := config.Route(config.AllPrimaries)
+	clusterScriptOptions.Route = route
```

Comment:
This should work
```suggestion
	clusterScriptOptions.Route = config.AllPrimaries
```
or
```suggestion
	clusterScriptOptions.Route = config.AllPrimaries.ToPtr()
```

Reply:
Updated.